File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(' ')).trim();
    }

    public static String logLevel(String logLine) {
        var end = logLine.indexOf("]");
        var level = logLine.substring(1,end).toLowerCase();
        System.out.println(level);
        return level;
    }

    public static String reformat(String logLine) {
        var level = logLevel(logLine);
        var message = message(logLine);
        return String.format("%s (%s)", message, level);
    }
}

File: fsharp/log-levels/LogLevels.fs
module LogLevels

let decompose (s:string) =
    match s.Split(": ") with 
    | [|level; message|] -> (level.Trim([|'[';']'|]).ToLower(), message.Trim())
    | _ -> failwith "unable to split"

let message (logLine: string): string = decompose logLine |> snd
    
let logLevel(logLine: string): string = decompose logLine |> fst

let reformat(logLine: string): string = sprintf "%s (%s)" (message logLine) (logLevel logLine)
